indexing-service/src/main/java/io/druid/indexing/overlord/TaskRunnerWorkItem.java

```diff
@@ -82,10 +83,7 @@ public DateTime getQueueInsertionTime()
     return queueInsertionTime;
   }
 
-  public TaskRunnerWorkItem withQueueInsertionTime(DateTime time)
-  {
-    return new TaskRunnerWorkItem(taskId, result, createdTime, time);
-  }
+  public abstract TaskLocation getLocation();
```

Comment:
that's a good question, I'm actually not really sure what the compareTo is supposed to be for. It already didn't include all the fields. will look into that a bit.